pkg/resources/manager/manager.go

```diff
@@ -131,38 +133,60 @@ func (r *resourceManager) setVMLimit(maxVMsLimit uint64) error {
 	return nil
 }
 
-func (r *resourceManager) createMachineClasses() error {
-	r.machineClasses = make([]*machineclass, len(r.tmpIRIMachineClasses))
-	for index, class := range r.tmpIRIMachineClasses {
-		resourceManagerClass := &machineclass{
-			// break reference
-			name:         class.GetName(),
-			capabilities: iri.MachineClassCapabilities{CpuMillis: class.Capabilities.CpuMillis, MemoryBytes: class.Capabilities.MemoryBytes},
-			resources: core.ResourceList{
-				core.ResourceCPU:    *resource.NewQuantity(class.Capabilities.CpuMillis, resource.DecimalSI),
-				core.ResourceMemory: *resource.NewQuantity(class.Capabilities.MemoryBytes, resource.BinarySI),
+func (r *resourceManager) initMachineClasses() error {
+	classes, err := loadMachineClassesFile(r.machineclassesFile)
+	if err != nil {
+		return err
+	}
+
+MAIN:
+	for _, class := range classes {
+		cpu, ok := class.Capabilities[core.ResourceCPU]
+		if !ok {
+			return fmt.Errorf("required resource %s is missing in machine class file", core.ResourceCPU)
+		}
+
+		mem, ok := class.Capabilities[core.ResourceMemory]
+		if !ok {
+			return fmt.Errorf("required resource %s is missing in machine class file", core.ResourceMemory)
+		}
+
+		// it is for minimize creation operations in status call
+		class.iriClass = &iri.MachineClass{
+			Name: class.Name,
+			Capabilities: &iri.MachineClassCapabilities{
+				CpuMillis:   cpu.MilliValue(),
+				MemoryBytes: mem.Value(),
 			},
 		}
 
+		for key := range class.Capabilities {
+			_, ok := r.registredResources[key]
```

Comment:
One of the keys in capabilities is `memory_bytes`, which is not a resource.

Reply:
hm? capabilities is resourcelist from machineclass file, it isn't iri.MachineClass